src/js/libgeo/GeoOps.js

```diff
@@ -1928,14 +2018,23 @@ geoMacros.Calculation = function(el) {
     return [];
 };
 
+geoMacros.Arc = function(el) {
+    el.type = "ArcBy3";
+    return [el];
+};
+
 geoMacros.Transform = function(el) {
-    var trKind = csgeo.csnames[el.args[0]].kind;
-    var argKind = csgeo.csnames[el.args[1]].kind;
+    var arg = csgeo.csnames[el.args[1]];
+    var tr = csgeo.csnames[el.args[0]];
+    // workaround for Arcs since we treat them as circles
+    var akind = arg.kind;
+    if (arg.kind === "C" && arg.type === "ArcBy3") akind = "Arc";
```

Comment:
> … the macros are evaluated before initialization takes place.

Then how about moving the `isArc` property from the geometric element to the geometric operator? So you can do `geoOps[arg.type].isArc`. There definitely should be a test case for repeated transformations, since this is more tricky than it would seem at first glance.

By the way, #144 will make sure that every element gets fully initialized before any subsequent macro gets expanded.